### ISSUE TITLE
Variadic Generic self invariance inconsistency/bug?

### ISSUE THREAD

**Issue by alwaysmpe** [microsoft/pyright#9340]
**Describe the bug**
Passing a variadic generic class to a constructor that takes that type with a variable length any tuple argument does not accept the self type but does accept the explicitly annotated class type or subclass type.
The error raised is that the type argument is invariant and does not match.

I would expect that either the self type is accepted, but if not then neither should the subclass type.

From the error message it looks like the type argument to the `Self` type isn't being treated as a `TypeVarTuple`.

**Code or Screenshots**

```python
from __future__ import annotations # bring on 3.14!
from typing import Any, Self

type ComposableVT = Composable[*tuple[Any, ...]]


class Composable[*SeqT]:

    sequence: tuple[*SeqT]

    def __init__(self, *args: *SeqT):
        self.sequence = args

    def __add__[*T](self, other: Composable[*T]) -> Sequence:
        # Argument of type "Self@Composable[SeqT@Composable]" cannot be assigned to parameter of type "ComposableVT"
        #   "Composable[SeqT@Composable]*" is not assignable to "Composable[*tuple[Any, ...]]"
        #     Type parameter "SeqT@Composable" is invariant, but "SeqT@Composable" is not the same as "*tuple[Any, ...]"  (reportArgumentType)
        return Sequence(self, other)

    def add_explicit[*T](self: Composable[*SeqT], other: Composable[*T]) -> Sequence:
        # No error
        return Sequence(self, other)

    def add_self[*T](self: Self, other: Composable[*T]) -> Sequence:
        # Argument of type "Self@Composable[SeqT@Composable]" cannot be assigned to parameter of type "ComposableVT"
        #   "Composable[SeqT@Composable]*" is not assignable to "Composable[*tuple[Any, ...]]"
        #     Type parameter "SeqT@Composable" is invariant, but "SeqT@Composable" is not the same as "*tuple[Any, ...]"  (reportArgumentType)
        return Sequence(self, other)

    def add_sequence(self: Composable[*SeqT], other: Sequence) -> Sequence:
        # No error
        return Sequence(self, other)


class Sequence(Composable[*tuple[ComposableVT, ComposableVT, *tuple[ComposableVT, ...]]]):
    """
    Simple sequence.
    """
```

**VS Code extension or command-line**
pyright cli, [pyright playground](https://pyright-play.net/?strict=true&code=GYJw9gtgBA%2BjwFcAuCQFM5QJYQA5hCSgEMA7UsJYpLMUgZygGIoAjELUgcyjqgGYAdAEYALAEIAsAChQkKEgCeuTjxz5CUAIKlFAGigBlNABtgMmUtxooAYUj56xVibQA1ACpQAvHYdgnFzQAbQAqFFxXYJ19KEF4gF0Ei2kZAGMTYnpGezwA5yjQ4wBHDwSALhSoaqh6NGKENFI0tHKFBEiQovqyqpqAEzRgWBhOLCQ4AAo6swNQ4hAuejbu0oBKSuka7drTYEE6hqaWnxJF%2Bj7qweG4Yn7%2BuDCy6b2DSgALNBA23McCrrKaygAFoAHxGeqNZqtGQ7GosLSLBAQJpEMDDKw2ABExjMAAFfvkgsESh4Cf5Aq4ElioGkyBQiKwbFl6FguKQ0P0FGAoLgFsQUUgvrwMcpsYTKe4PFjYXDmDUsRL-iSeuS8pKEqEaVhGAySNk2aR-tyoIqKcrwh0ojEDPFBEkZVs5Sxth4xbz%2BYLhTjVUqgtrGJwAG4LLBkJAGVjIU2ktV-f3YXWUBSfWoC5mMLGWzrRXS2xI0qCTdAaJCIrjI1Fu6xrWVw9AoECkCFHaEvWa8JCfEC11JOq5DEj3GBoAAekSwaXGTwS7eAP3NxNWZTeXa%2BC-VFsBIPBJShLU2cvlADkeV9wCA6zsG6hm3vjmg56vu72r9chw8ZsAZ3O2rjgM%2B65%2BJuS7bmCLb7jC-Y7AiSIoqQaKitYMZ7HGRJRLGfpUjSdLkMmTL6qy7KciafIgOmQogCKCjumaIGuJ4jpHi6ppYSEmGLlSWqJlAeosoaxpIDydHxoUETWnmcSJNSV7bCx1TVjYZEUd6HH0WgAbYKQIYcOGkbRj6pRoZKml6muaYovqprZhJsR2tS1TFmgpblpWCGKb2R43k2EEPk%2BnYvpcUDvncn6Qn5X4bqJXSkgkgHfL50JAuB97QoezpQKeUDngQsk1N5d7hW2X7xa%2BfYZCyiUtJMbFhOJIRsZ4BiNR4cz1cELX5vaSQbFeWL9VehjqK4uyti0gh9f1QA), latest

**Comment by erictraut** [comment]
I agree this is a bug. Thanks for reporting it.

It will be addressed in the next release.

**Comment by alwaysmpe** [comment]
Awesome thanks, mildly related #9343 (when I actually ran my code it crashed anyway so there's that)

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.387.
